recipes/liboping/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS} oping)
```

```suggestion
include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

find_package(liboping CONFIG REQUIRED)

add_executable(${PROJECT_NAME} test_package.c)
target_link_libraries(${PROJECT_NAME} liboping::liboping)
```

Generated targets from Conan.

There is no pkg-config or cmake file for this file, however, there is a spec file: https://github.com/octo/liboping/blob/master/contrib/liboping.spec, which uses liboping for packaging, so I believe `Findliboping.cmake` is the correct way.

---

This totally busted my test build 

Scanning dependencies of target test_package
[ 50%] Building C object CMakeFiles/test_package.dir/test_package.c.o
[100%] Linking C executable bin/test_package
/usr/bin/ld: CMakeFiles/test_package.dir/test_package.c.o: in function `main':
test_package.c:(.text.startup+0x15): undefined reference to `ping_construct'
/usr/bin/ld: test_package.c:(.text.startup+0x22): undefined reference to `ping_destroy'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/test_package.dir/build.make:84: bin/test_package] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/test_package.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
ERROR: liboping/1.10.0 (test package): Error in build() method, line 12
	cmake.build()
	ConanException: Error 2 while executing cmake --build '/home/maurice/wti/conan-center-index/recipes/liboping/all/test_package/build/878b1b6e2d4cbc82452d014e37c38868584457e8' '--' '-j20'

---

it doesn't pick up the liboping.a in the linker config

---

It works for CI, we need more details to understand your scenario. What profile did you use, OS, full log ...

---

@uilianries  If doesn't work for CI. I did commit the changes and the latest CI build is still in failed state - please look at the result of the Linux GCC compile, specifically the test_package section.

---

I see, I'll take a look, thanks!

---

@uilianries  Did I possibly expose an issue in the CMake macro? So while the easiest fix would be to revert the changes - maybe it's better to fix the macro?

---

@MauriceS-WTI please, take a look on my new suggestion, we forgot a basic information 😅